lib/ruby_lsp/server.rb

```diff
@@ -188,6 +193,17 @@ module RubyLsp
       end
     end
 
+    on("textDocument/inlayHint", parallel: true) do |request|
+      document = store.get(request.dig(:params, :textDocument, :uri))
+      range = request.dig(:params, :range)
+      start_line = range.dig(:start, :line)
+      end_line = range.dig(:end, :line)
+
+      if document.parsed?
+        Requests::InlayHints.new(document, start_line..end_line).run
+      end
```

Comment:
```suggestion
      return unless document.parsed?
      
      Requests::InlayHints.new(document, start_line..end_line).run
```

Reply:
You can't directly from this block or it results in a `LocalJumpError`. We've been bitten by this before 😕

Reply:
But you can do `next unless document.parsed?` 😉

Reply:
Urgh

Reply:
On this point, I actually want to make the LSP gem more typed so that instead of registering blocks, you start overriding methods that know what parameters of what type will be passed in for the request that you are handling. See:
- https://github.com/mtsmfm/language_server-protocol-ruby/issues/46, and
- https://github.com/mtsmfm/language_server-protocol-ruby/issues/47